### ISSUE TITLE
wires connects to exsiting Firefox session rather than creating a new one

### ISSUE THREAD

**Issue by EricRahm** [mozilla/geckodriver#42]
STR:
1) Open Firefox
2) Run snippet below
3) Note that a new Firefox instance was not opened, the page was loaded in the existing instance

```
from selenium import webdriver
from selenium.webdriver.common.desired_capabilities import DesiredCapabilities

firefox_capabilities = DesiredCapabilities.FIREFOX.copy()
firefox_capabilities['marionette'] = True
firefox_capabilities['binary'] = '/Applications/FirefoxNightly.app/Contents/MacOS/firefox'
driver = webdriver.Firefox(capabilities=firefox_capabilities)

driver.get('data:text/html,<a href="https://mozilla.org">Click This!</a>')
```

**Comment by jgraham** [comment]
I can't reproduce this. @AutomatedTester can you?

**Comment by EricRahm** [comment]
This was on OS X in case that wasn't clear. It appears that sometimes a new browser is opened, but the page is still loaded in the other instance.

**Comment by EricRahm** [comment]
@jgraham I think I have better steps to repro:
# Open Nightly <---- has to be nightly, it has marionette enabled it seems
# Run snippet

So I get the feeling wires will open up Fx, tries to bind to the default port but the existing Nightly already has it, then happily connects to the existing session.

**Comment by titusfortner** [comment]
I'm reproducing this on Linux, Firefox 45 right now. v0.6.2

**Comment by andreastt** [comment]
This implies the Nightly process starts with Marionette enabled and bound to port 2828 by default. This is not the case for Nightly on my system:

```
% lsof -i :2828
% 
```

**Comment by martijnhjk** [comment]
Any updates on this? I am getting the same behaviour.

Initial test runs, subsequent tests fail (regardless the previous passed or failed).

-------------- Stacktrace from my Eclipse IDE (for subsequent tests): -------------- 

org.openqa.selenium.remote.UnreachableBrowserException: Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure.
Build info: version: '2.53.0', revision: '35ae25b1534ae328c771e0856c93e187490ca824', time: '2016-03-15 10:43:46'
System info: host: 'Martijns-MacBook-Pro.local', ip: '192.168.1.116', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.11.5', java.version: '1.8.0_25'
Driver info: driver.version: MarionetteDriver

-------------- Marionette debug log from initial test: -------------- 

1466098150020	Marionette	DEBUG	Marionette enabled via build flag and pref
1466098150129	Marionette	INFO	Listening on port 2828
1466098150240	Marionette	DEBUG	Marionette enabled via command-line flag
1466098150344	Marionette	DEBUG	Accepted connection conn0 from 127.0.0.1:57717
1466098150380	Marionette	TRACE	conn0 -> [0,1,"newSession",{"capabilities":{"desiredCapabilities":{"browserName":"firefox","platform":"ANY","version":""},"requiredCapabilities":{}},"sessionId":null}]
1466098150381	Marionette	CONFIG	Changing capabilities: {"browserName":"Firefox","browserVersion":"47.0","platformName":"Darwin","platformVersion":"15.5.0","specificationLevel":0,"raisesAccessibilityExceptions":false,"rotatable":false,"acceptSslCerts":false,"takesElementScreenshot":true,"takesScreenshot":true,"proxy":{},"platform":"DARWIN","XULappId":"{ec8030f7-c20a-464f-9b0e-13a3a9e97384}","appBuildId":"20160604131506","device":"desktop","version":""}

-------------- last two lines in my logs: -------------- 

1466098180167	Marionette	TRACE	conn0 -> [0,117,"quitApplication",{"flags":["eForceQuit"]}]
1466098180167	Marionette	TRACE	conn0 <- [1,117,null,{}]

I have driver.quit on my @After to shut down the driver after each test.

any help much appreciated!

**Comment by brtsay** [comment]
I am also experiencing this issue on Ubuntu 14.04, Firefox 47, geckodriver-0.8.0-linux64

**Comment by cam72cam** [comment]
I am also experiencing this issue on OSX 10.10.5, Firefox 47, geckodriver-0.8.0

**Comment by jgraham** [comment]
I think this is fixed by #112, please reopen if not.
